### ISSUE TITLE
Add a synchronous adapter

### ISSUE THREAD

**Issue by SamMousa** [reactphp/filesystem#87]
While I recognize that the primary purpose of `reactphp/filesystem` is to have non-blocking io, it does have some other benefits:

- Event / promise based
- Abstraction from filesystem functions

In some cases the potential gains from non-blocking io are not worth guaranteed costs. In such scenarios it would still be nice to have a synchronous adapter that allows us to use the same code.

From a consumer point of view as long as the functions return promises it doesn't matter if it is blocking or not.

**Comment by WyriHaximus** [comment]
> From a consumer point of view as long as the functions return promises it doesn't matter if it is blocking or not.

From anything else inside the event loop is does matter, as they will be blocked by any blocking operation. And that is something we don't support in any of our packages.

**Comment by SamMousa** [comment]
> And that is something we don't support in any of our packages.

Fair enough, if that's the choice that has been made no there's no use discussing the worth of synchronous implementation(s).
